File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s936984238_commented.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	for {
		var n int
		fmt.Scanf("%d", &n)
		if n == 0 {
			break
		}
		scores := make([]int, n)
		for i := 0; i < n; i++ {
			fmt.Scanf("%d", &scores[i])
		}
		sort.Ints(scores)
		sum := 0
		for i := 1; i < n-1; i++ {
			sum += scores[i]
		}
		fmt.Println(sum / (n-2))
	}
}// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s936984238_commented.py
import sys
data = list(map(int, sys.stdin.read().split()))
idx = 0
while True:
    number = data[idx]
    idx += 1
    if number == 0:
        break
    scores = data[idx:idx+number]
    idx += number
    scores.sort()
    s = sum(scores[1:-1])
    print(s // (number-2))
# End of Code
